segmentation.py

```diff
@@ -0,0 +1,575 @@
+"""Segment time sequence data.
+
+The Bayesian change-point detection model (BCDM) class implements a recursive
+algorithm for partitioning a sequence of real-valued input-output data into
+non-overlapping segments. The segment boundaries are chosen under the
+assumption that, within each segment, the data follow a multi-variate linear
+model.
+
+Segmentation is carried out in an online fashion by recursively updating a set
+of hypotheses. The hypotheses capture the belief about the current segment,
+e.g. its duration and the linear relationship between inputs and outputs, given
+all the data so far. Each time a new pair of data is received, the hypotheses
+are propagated and re-weighted to reflect this new knowledge.
+
+.. codeauthor:: Gabriel Agamennoni <abriel.agamennoni@mavt.ethz.ch>
+.. codeauthor:: Asher Bender <a.bender@acfr.usyd.edu.au>
+
+"""
+import numpy as np
```

Comment:
It is not PEP8 compliance. It is common practice supported by (originating from?) many of the large projects. For instance [documentation in SciPy](http://scipy.org/getting-started.html#an-example-script). The [pylab option](http://ipython.org/ipython-doc/2/api/generated/IPython.core.magics.pylab.html#IPython.core.magics.pylab.PylabMagics.pylab) in ipython imports numpy as np. 

That said, it is not a requirement. If you prefer longer namespacing I can get onboard with that! Although if you do use ipython (I recommend it) **np** is good for the pylab option.